### PR TITLE
Allow to build library without examples or tests.

### PR DESCRIPTION
**Pull request by archont94**
By setting option WITH_EXAMPLES to OFF, we can skip this build step.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -525,175 +526,179 @@ SET_TARGET_PROPERTIES(vncclient vncserver
...
-if(APPLE AND NOT IOS AND WITH_THREADS AND CMAKE_USE_PTHREADS_INIT)
-  set(LIBVNCSERVER_EXAMPLES
-    ${LIBVNCSERVER_EXAMPLES}
-    mac
-  )
-  find_library(CARBON_LIBRARY Carbon)
-  find_library(IOKIT_LIBRARY IOKit)
-  find_library(IOSURFACE_LIBRARY IOSurface)
-endif(APPLE AND NOT IOS AND WITH_THREADS AND CMAKE_USE_PTHREADS_INIT)
+  if(WITH_THREADS AND (CMAKE_USE_PTHREADS_INIT OR CMAKE_USE_WIN32_THREADS_INIT))
```
Comment (bk138) [inline CMakeLists.txt:550]: please make sure only relevant lines are changed.

**Review by bk138 — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (archont94) [reply CMakeLists.txt]: Done.

File: CMakeLists.txt [hunk 2]
```diff
@@ -604,96 +606,100 @@ if(FFMPEG_FOUND)
...
-endforeach(e ${LIBVNCSERVER_EXAMPLES})
-
-foreach(e ${LIBVNCCLIENT_EXAMPLES})
-  add_executable(client_examples_${e} ${LIBVNCCLIEXAMPLE_DIR}/${e}.c ${LIBVNCCLIEXAMPLE_DIR}/${${e}_EXTRA_SOURCES} )
-  set_target_properties(client_examples_${e} PROPERTIES OUTPUT_NAME ${e})
-  set_target_properties(client_examples_${e} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/client_examples)
-  target_link_libraries(client_examples_${e} vncclient ${CMAKE_THREAD_LIBS_INIT} ${SDL2_LIBRARY} ${GTK2_LIBRARIES} ${FFMPEG_LIBRARIES})
-endforeach(e ${LIBVNCCLIENT_EXAMPLES})
-
+if(WITH_EXAMPLES)
```
Comment (bk138) [inline CMakeLists.txt:609]: that's nice now :+1:

File: CMakeLists.txt [hunk 3]
```diff
@@ -58,6 +58,8 @@ option(WITH_24BPP "Allow 24 bpp" ON)
 option(WITH_IPv6 "Enable IPv6 Support" ON)
 option(WITH_WEBSOCKETS "Build with websockets support" ON)
 option(WITH_SASL "Build with SASL support" ON)
+option(WITH_EXAMPLES "Build examples" ON)
+option(WITH_TEST "Build test" ON)
```
Comment (bk138) [inline CMakeLists.txt:62]: please name this WITH_TESTS

File: CMakeLists.txt [hunk 4]
```diff
@@ -604,96 +606,100 @@ if(FFMPEG_FOUND)
...
+endif(WITH_EXAMPLES)
 
 #
 # them tests
 #
 
-if(UNIX)
-    set(ADDITIONAL_TEST_LIBS m)
-endif(UNIX)
+if(WITH_TEST)
```
Comment (bk138) [inline CMakeLists.txt:629]: that diff looks huge. please only add the `if(WITH_TESTS)` and `endif(WITH_TESTS)` lines. i take respsonsibility for the wrong indent of contained lines ;-)

**Review by bk138 — CHANGES_REQUESTED** [review]
thanks for the adaptations. few more comments, than we're good to go!

File: CMakeLists.txt (hunk 3 above)
Comment (archont94) [reply CMakeLists.txt]: Changed.

File: CMakeLists.txt (hunk 2 above)
Comment (archont94) [reply CMakeLists.txt:609]: 👍

File: CMakeLists.txt (hunk 4 above)
Comment (archont94) [reply CMakeLists.txt]: Changed :).

**Pull request merged**